runtime/src/bank/fee_distribution.rs

```diff
@@ -0,0 +1,953 @@
+        assert_eq!(transaction_fees, bank.collector_fees.load(Relaxed));
+
+        let initial_capitalization = bank.capitalization();
+        let initial_collector_id_balance = bank.get_balance(bank.collector_id());
+        bank.distribute_transaction_fees();
+        let new_collector_id_balance = bank.get_balance(bank.collector_id());
+
+        assert_eq!(initial_collector_id_balance, new_collector_id_balance);
+        assert_eq!(
+            initial_capitalization - transaction_fees,
+            bank.capitalization()
+        );
+        let locked_rewards = bank.rewards.read().unwrap();
+        assert!(
+            locked_rewards.is_empty(),
+            "There should be no rewards distributed"
+        );
+    }
+
+    #[test]
+    fn test_distribute_transaction_fees_overflow_failure() {
+        let genesis = create_genesis_config(0);
+        let bank = Bank::new_for_tests(&genesis.genesis_config);
+        let transaction_fees = 100;
+        bank.collector_fees.fetch_add(transaction_fees, Relaxed);
+        assert_eq!(transaction_fees, bank.collector_fees.load(Relaxed));
+
+        // ensure that account balance will overflow and fee distribution will fail
+        let account = AccountSharedData::new(u64::MAX, 0, &system_program::id());
+        bank.store_account(bank.collector_id(), &account);
+
+        let initial_capitalization = bank.capitalization();
+        let initial_collector_id_balance = bank.get_balance(bank.collector_id());
+        bank.distribute_transaction_fees();
+        let new_collector_id_balance = bank.get_balance(bank.collector_id());
+
+        assert_eq!(initial_collector_id_balance, new_collector_id_balance);
+        assert_eq!(
+            initial_capitalization - transaction_fees,
+            bank.capitalization()
+        );
+        let locked_rewards = bank.rewards.read().unwrap();
+        assert!(
+            locked_rewards.is_empty(),
+            "There should be no rewards distributed"
+        );
+    }
+
+    #[test]
+    fn test_deposit_fees() {
+        let initial_balance = 1_000_000_000;
+        let genesis = create_genesis_config(initial_balance);
+        let bank = Bank::new_for_tests(&genesis.genesis_config);
+        let pubkey = genesis.mint_keypair.pubkey();
+
+        let deposit_amount = 500;
+        let options = DepositFeeOptions {
+            check_account_owner: true,
+            check_rent_paying: true,
+        };
+
+        // Call the deposit_fees function
+        let result = bank.deposit_fees(&pubkey, deposit_amount, options);
+
+        // Check the result
+        match result {
+            Ok(new_balance) => {
+                assert_eq!(
+                    new_balance,
+                    initial_balance + deposit_amount,
+                    "New balance should be the sum of the initial balance and deposit amount"
+                );
+            }
+            Err(e) => panic!("Unexpected error: {:?}", e),
```

Comment:
Unwrapping above would print the error, wouldn't it? I feel like the match syntax doesn't gain us much here.